common/models/queue.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+const { getPlaylist, updatePlaylist } = require('../../server/utils/playlist');
+const { addToDefaultSongs } = require('../../server/utils/adminQueue');
+const { playCurrentSong, pauseCurrentSong } = require('../../server/utils/player');
+
+module.exports = function (Queue) {
+  Queue.getPlaylist = function (id, cb) {
+    getPlaylist(id)
+      .then((tracks) => cb(null, tracks))
+      .catch(err => cb(err));
+  }
+
+  Queue.remoteMethod('getPlaylist', {
+    description: 'Gets current playlist from Spotify',
+    accepts: {
+      arg: 'id',
+      type: 'string'
+    },
+    http: {
+      path: '/getPlaylist',
+      verb: 'get'
+    },
+    returns: {
+      arg: 'data',
+      type: 'array',
+      root: true
+    },
+  });
+
+  Queue.updatePlaylist = function (id, songID, cb) {
+    updatePlaylist(id, songID)
+      .then((queue) => {
+        Queue.replaceOrCreate(queue)
```

Comment:
Loopback's callback should be invoked after replaceOrCreate has completed. Replace this line with:

```suggestion
        Queue.replaceOrCreate(queue, cb)
```